### PR TITLE
test: Test for Verify flow

### PR DESCRIPTION
**Pull request by bistaastha**
## Code Coverage
Screenshot 2024-11-27 at 11 46 00 PM

## Summary by CodeRabbit

- **New Features**
	- Enhanced user interaction with a new back navigation method.
	- Improved password validation with dynamic checks and feedback.
	- Conditional rendering for form elements to ensure clarity and usability.
	- New error messages for password fields to provide specific feedback.
	- Dynamic updates to password criteria in the tooltip component.
	- Updated layout and structure for the account setup interface.

- **Bug Fixes**
	- Resolved issues with password matching validation and form control handling.

- **Documentation**
	- Updated test suite for better readability and coverage of password validation scenarios.

- **Chores**
	- Streamlined initialization and handling of password criteria in the tooltip component.
	- Added new styling for error handling in the user interface.
	- Comprehensive updates to the styling and structure of the account setup component.

### CODE REVIEW

**Review by Chethan-Fyle — APPROVED** [review]

File: src/app/post-verification/invited-user/invited-user.page.html [hunk 1]
```diff
@@ -3,12 +3,12 @@
...
       class="fy-icon back-icon"
       src="../../../assets/svg/arrow-tail-left.svg"
-      [routerLink]="['/','auth','sign_in']"
+      (click)="redirectToSignIn()"
     ></ion-icon>
     <div>
       <div class="invited-user__form-header">Join "{{ orgName }}" on Fyle</div>
       <div>
-        <form [formGroup]="fg">
+        <form *ngIf="fg" [formGroup]="fg">
```
[the remark this answers is not in the document]
Comment (bistaastha) [reply src/app/post-verification/invited-user/invited-user.page.html:11]: Not needed here

File: src/app/post-verification/invited-user/invited-user.page.ts [hunk 2]
```diff
@@ -66,8 +66,8 @@
 
     this.fg = this.fb.group({
       fullName: ['', Validators.compose([Validators.required, Validators.pattern(/[A-Za-z]/)])],
-      password: ['', [Validators.required, this.customPasswordValidator()]],
-      confirmPassword: ['', [Validators.required, this.passwordMatchValidator()]],
+      password: ['', [Validators.required, this.customPasswordValidator]],
+      confirmPassword: ['', [Validators.required, this.validatePasswordEquality]],
```
[the remark this answers is not in the document]
Comment (bistaastha) [reply src/app/post-verification/invited-user/invited-user.page.ts:70]: not needed

File: src/app/post-verification/invited-user/invited-user.page.ts [hunk 3]
```diff
@@ -94,8 +94,8 @@
           switchMap(() => this.eou$),
           switchMap((eou) => {
             const user = eou.us;
-            user.full_name = this.fg.controls.fullName.value;
-            user.password = this.fg.controls.password.value;
+            user.full_name = this.fg.controls.fullName.value as string;
+            user.password = this.fg.controls.password.value as string;
```
[the remark this answers is not in the document]
Comment (bistaastha) [reply src/app/post-verification/invited-user/invited-user.page.ts:98]: password will always be string

File: src/app/post-verification/invited-user/invited-user.page.spec.ts [hunk 4]
```diff
@@ -270,17 +267,48 @@
...
+      component.setPasswordTooltip(false);
+      expect(component.showPasswordTooltip).toBeFalse();
+    });
+  });
+
+  it('redirectToSignIn(): should navigate to the sign-in page', () => {
+    component.redirectToSignIn();
+    // @ts-ignore
+    expect(component.router.navigate).toHaveBeenCalledOnceWith(['/', 'auth', 'sign_in']); // Should navigate to the correct route
+  });
```
[the remark this answers is not in the document]
Comment (bistaastha) [reply src/app/post-verification/invited-user/invited-user.page.spec.ts:313]: This is needed

File: src/app/shared/components/password-check-tooltip/password-check-tooltip.component.scss [hunk 5]
```diff
@@ -14,16 +16,16 @@
...
     border-radius: 8px;
-    background: $white;
+    background: $pure-white;
 
     &::before {
       position: absolute;
-      bottom: 100%;
-      left: 50%;
+      bottom: 100%; // Position the arrow above the tooltip box
+      left: 10%;
```
[the remark this answers is not in the document]
Comment (bistaastha) [reply src/app/shared/components/password-check-tooltip/password-check-tooltip.component.scss:24]: it is supposed to be positioned to the left only

**Pull request merged**